### PR TITLE
Improve toaster management

### PR DESCRIPTION
**Pull request by cedricpim**

### CODE REVIEW

File: src/ui/ToasterManager.js [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+  const dispatch = useDispatch()
 
   function dismiss(key, toasters) {
+    dispatch(dismissFromRedux(key))
     const entry = toasters.find((entry) => entry.key === key)
     if (toasters.indexOf(entry) === -1) {
       return
     }
 
+    console.log('Dismissing key', key, [...toasters])
```
Comment (NoryDev) [inline src/ui/ToasterManager.js:20]: do we want to console.log stuff?

**Review by NoryDev — APPROVED** [review]
apart from the console.log comment, that looks good.

I tried to compare with frontend-react to see that all was ported, but since there is quite a lot, I hope I didn't miss anything.

Did you deviate in any way from the version that is in frontend-react? Or is it an exact port?

File: src/ui/ToasterManager.js (hunk 1 above)
Comment (cedricpim) [reply src/ui/ToasterManager.js:20]: Probably not, I will clean this up :+1:

**Comment by cedricpim** [comment]
> apart from the console.log comment, that looks good.
> 
> I tried to compare with frontend-react to see that all was ported, but since there is quite a lot, I hope I didn't miss anything.
> 
> Did you deviate in any way from the version that is in frontend-react? Or is it an exact port?

Deviated a little bit, since I made ExportComponent have a store argument. A react store is something that doesn't seem to make sense in a library but could be passed on from outside.

**Comment by NoryDev** [comment]
> > apart from the console.log comment, that looks good.
> > I tried to compare with frontend-react to see that all was ported, but since there is quite a lot, I hope I didn't miss anything.
> > Did you deviate in any way from the version that is in frontend-react? Or is it an exact port?
> 
> Deviated a little bit, since I made ExportComponent have a store argument. A react store is something that doesn't seem to make sense in a library but could be passed on from outside.

Ah yeah, maybe a bit weird, is it easy to make the store an argument?

**Comment by cedricpim** [comment]
> > > apart from the console.log comment, that looks good.
> > > I tried to compare with frontend-react to see that all was ported, but since there is quite a lot, I hope I didn't miss anything.
> > > Did you deviate in any way from the version that is in frontend-react? Or is it an exact port?
> > 
> > 
> > Deviated a little bit, since I made ExportComponent have a store argument. A react store is something that doesn't seem to make sense in a library but could be passed on from outside.
> 
> Ah yeah, maybe a bit weird, is it easy to make the store an argument?

Yes, it works as any other arguments. We just need to provide it once, on setting the ExportedComponent. I will do an MR frontend-react repository with the changes :+1:

**Pull request merged**